core/src/test/java/de/evoila/cf/broker/service/impl/AsyncDeploymentServiceImplTest/AsyncDeleteInstanceTest.java

```diff
@@ -2,57 +2,60 @@
 
 import de.evoila.cf.broker.exception.ServiceBrokerException;
 import de.evoila.cf.broker.model.JobProgress;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.*;
 
-class AsyncDeleteInstanceTest extends BaseTest
-{
-    private void whensForSyncDeleteInstanceThrowsException(JobProgress returnOfFailJob) throws ServiceBrokerException
-    {
+class AsyncDeleteInstanceTest extends BaseTest {
+
+    private static final String JOB_PROGRESS_DESCRIPTION = "Deleting service..";
+
+    private void whenSyncDeleteInstanceThrowsException(Exception expectedException) throws ServiceBrokerException {
         mockSuccessfulStartJob(JobProgress.DELETE);
-        ServiceBrokerException expectedException = new ServiceBrokerException("Test");
         doThrow(expectedException).when(deploymentService).syncDeleteInstance(serviceInstance, plan, platformService);
-        when(startedJob.getId())
-                .thenReturn(JOB_PROGRESS_ID);
-        when(jobProgressService.failJob(eq(JOB_PROGRESS_ID), anyString()))
-                .thenReturn(returnOfFailJob);
     }
 
-    @Test
-    void startJobReturnsNull()
-    {
-        when(serviceInstance.getId())
-                .thenReturn(SERVICE_INSTANCE_ID);
-        when(jobProgressService.startJob(eq(JOB_PROGRESS_ID), eq(SERVICE_INSTANCE_ID), anyString(), eq(JobProgress.DELETE)))
-                .thenReturn(null);
+    private void whenSyncDeleteInstanceThrowsServiceBrokerException() throws ServiceBrokerException {
+        whenSyncDeleteInstanceThrowsException(new ServiceBrokerException("Test"));
+    }
+
+    private void whenSyncDeleteInstanceThrowsRuntimeException() throws ServiceBrokerException {
+        whenSyncDeleteInstanceThrowsException(new RuntimeException("Test"));
+    }
 
+    @Test
+    @DisplayName("Should log exception, when jobStart(...) throws ServiceBrokerException.")
+    void startJobThrowsServiceBrokerException() throws ServiceBrokerException {
+        mockStartJobThrowsServiceBrokerException(JobProgress.DELETE, JOB_PROGRESS_DESCRIPTION);
         asyncDeploymentService.asyncDeleteInstance(null, serviceInstance, null, null, JOB_PROGRESS_ID);
     }
 
     @Test
-    void syncDeleteInstanceThrowsExceptionFailJobReturnsNull() throws ServiceBrokerException {
-        whensForSyncDeleteInstanceThrowsException(null);
+    @DisplayName("Should log exception and update JobProgress, when jobStart(...) throws RuntimeException.")
+    void startJobThrowsRuntimeException() throws ServiceBrokerException {
+        mockStartJobThrowsRuntimeException(JobProgress.DELETE, JOB_PROGRESS_DESCRIPTION);
+        asyncDeploymentService.asyncDeleteInstance(null, serviceInstance, null, null, JOB_PROGRESS_ID);
```

Comment:
Are the null parameters needed? Why not pass the mocked objects as in other tests?

Reply:
fixed